### PR TITLE
Rename `load_file_list_of_signatures` to `load_pathlist_from_file`

### LINKED ISSUE #1369: rename `load_file_list_of_signatures` in sourmash_args.py
The function `load_file_list_of_signatures` in `src/sourmash/sourmash_args.py` was initially used only for loading a text file that contained a list of signature filenames, but since has changed (e.g. #1362) to support lists of FASTA files.

TODO:
* rename the function to `load_pathlist_from_file` (or a better name if you have one :)
* find all places where the old name is used in the code base, perhaps with `git grep -l load_file_list_of_signatures` and rename it there
* optionally, add some Python unit tests in `tests/test_sourmash.py` for this function, including empty file list, badly formatted ones, ones with duplicates, ???

### PR DESCRIPTION
**Pull request by keyabarve**
Fixes #1369 

This PR renames the function ```load_file_list_of_signatures``` to ```load_pathlist_from_file``` in 6 files. These files are:
- ```src/sourmash/lca/command_summarize.py```
- ```src/sourmash/command_sketch.py```
- ```src/sourmash/commands.py```
- ```src/sourmash/lca/command_classify.py```
- ```src/sourmash/lca/command_index.py```
- ```src/sourmash/sourmash_args.py```

TODO:
- [ ] Add some Python unit tests in ```tests/test_sourmash.py``` for this function, including empty file list, badly formatted ones, ones with duplicates.

Checklist:
- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Please note your ORCID in the comments. If you don't have one, you can register for one [here](https://orcid.org/register).

### CODE REVIEW

**Comment by keyabarve** [comment]
Please review @ctb @luizirber

**Comment by ctb** [comment]
hi @keyabarve a couple things -

* first, it looks like you started from your last PRs branch, rather than the `latest` branch; so this needs to be updated from latest again :)
* separately, I'd suggest waiting to ask for a review _after_ the checks above run the tests successfully...
* last but not least, please put `Fixes #1369` in the pull request comment *body*, not the title - that's so that it's properly linked to the issue and the issue will be closed when this is merged. See comments at bottom https://github.com/dib-lab/sourmash/pull/1422 for more detailed instructions..

I note that @markjih is also working on this same issue, in https://github.com/dib-lab/sourmash/pull/1382, but he didn't link it to the issue so you couldn't know that. That PR is a bit stale now, so I'm fine with you continuing here :). In general, tho, the goal is to try to avoid stepping on each other, which is why good issue/PR linking is so important!

File: src/sourmash/lca/command_summarize.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def summarize_main(args):
     inp_files = args.query
 
     if args.query_from_file:
-        more_files = sourmash_args.load_file_list_of_signatures(args.query_from_file)
+        more_files = sourmash_args.load_pathlist_from_file(args.query_from_file)
```
Comment (ctb) [inline src/sourmash/lca/command_summarize.py:184]: please create a new issue for testing `--query-from-file` for `sourmash lca summarize` - thanks!

Comment (ctb) [reply src/sourmash/lca/command_summarize.py:184]: (because our current tests don't execute this, apparently!)

**Comment by keyabarve** [comment]
Will do, thanks!

**Comment by keyabarve** [comment]
@ctb Should I create a new branch or merge the RenameIssue1369 branch with some other?

**Comment by keyabarve** [comment]
Please review. @ctb @luizirber

**Comment by ctb** [comment]
hi @keya, the code changes look good. But let's work on improving the pull request title and description ;).

First, the title of the PR needs to be something that we can look at later to understand what this PR was about. This is because these remain associated with the project forever.  Right now the title is "Rename issue #1369" which isn't very informative - can you describe briefly what you did?, e.g. "rename `load_file_list_of_signatures` to ..."?

Second, the body of the PR description at the top needs to reference relevant issues (which it does!) as well as indicating to GitHub which issues this PR will close. In this case I _think_ this closes #1369, in which case you need to put the literal exact words "Fixes #1369" in there; then, when this PR is closed, issue #1369 will be closed automatically.

Third, the PR description should be at least somewhat informative. Yes, it's always possible to go read the code diff to figure out what was changed, but it's nice not to have to. So something like "This PR renames function XYZ to function ABC in files 1, 2, and 3." would be sufficient.

An example is [here](https://github.com/dib-lab/sourmash/pull/1406), if you'd like to see how I structure my PRs descriptions. You don't have to write a novel like I sometimes do, but in general any relevant information should be written in the PR description

Can you take a crack at updating the title and PR description, please? Thanks!

**Comment by keyabarve** [comment]
@ctb I have updated the title and PR description. Please review. 
Issue #1369 also mentions something about adding some more tests. Could you please elaborate on that a little?
Thanks!

**Review by ctb — APPROVED** [review]

**Comment by ctb** [comment]
thanks, @keyabarve, everything but the title is good now. Before merging, could you change the actual title of the PR (currently `Rename Issue #1369`) to something like `rename load_file_list_of_signatures to ...`?

Regarding adding some unit tests, I think that's best done as a separate PR at this point. Can you create a new issue (like #1427) for that, and then you can choose to take it as a new PR if you like.

in terms of actual tests, please see [this example](https://github.com/dib-lab/sourmash/pull/1406/files#diff-424b9a1a321f8bd96165987fd52d25996d315d8322cabeb0ae7711cd45b8a183R660) for the kind of thing you want to do - create a list of files that's broken in various ways, and then use the actual function `load_pathlist_from_file` to try loading and assert that what happens is correct (an error, or something else). I would suggest just getting started with _something_ and then I can help suggest more tests to add! The key point is that the tests should cover every explicit and implicit branch (if/else, try/except) in the function.

**Comment by ctb** [comment]
thank you!

**Pull request merged**